apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerParameter.cs

```diff
@@ -95,25 +129,23 @@ public override string ParameterName
         /// </summary>
         public SpannerDbType SpannerDbType
         {
-            get { return TypeCode.GetSpannerDbType(); }
-            set { TypeCode = value.GetSpannerTypeCode(); }
+            get;
+            set;
         }
 
         /// <inheritdoc />
         public override object Value
         {
-            get { return _value; }
+            get => _value;
             set
             {
-                if (TypeCode == TypeCode.Unspecified)
+                if (SpannerDbType.TypeCode == TypeCode.Unspecified && value != null)
```

Comment:
typically a parameter is fully set up with a type and then only when you are executing commands do you fill it in with a value.  So in this case, its completely reasonable to enforce that the type should have been set already (probably in its constructor).